addons/sale_timesheet/models/sale_order.py

```diff
@@ -78,6 +79,29 @@ class SaleOrderLine(models.Model):
 
     qty_delivered_method = fields.Selection(selection_add=[('timesheet', 'Timesheets')])
     analytic_line_ids = fields.One2many(domain=[('project_id', '=', False)])  # only analytic lines, not timesheets (since this field determine if SO line came from expense)
+    delivery_info = fields.Char(compute='_compute_delivery_info')
+
+    def name_get(self):
+        res = super(SaleOrderLine, self).name_get()
+        if self.env.context.get('with_remaining_hours'):
+            for index, name_get in enumerate(res):
```

Comment:
for perf (don't need filtered for each line)
```python
if self.env.context.get('with_remaining_hours'):
    names = dict(res)
    result = []
    for line in self:
         name = names.get(line.id)
         if line.product_id.service_policy == 'ordered_timesheet':
               name = name = '{name} - {delivery_info}'.format(name=name, delivery_info':line.delivery_info)
         result.append((line.id, name)
     return result
```

Reply:
👍